internal/instance/resource_instance.go

```diff
@@ -795,6 +871,13 @@ func (r InstanceResource) SyncState(ctx context.Context, tfState *tfsdk.State, s
 		return respDiags
 	}
 
+	if !m.SourceFile.IsNull() && !m.Devices.IsNull() {
```

Comment:
But shouldn't be the devices already be created before and returned from the instance state? Why is this not the case in this scenario?

Reply:
My understanding is, that in this case, we do not actually define a device but we only use the configuration section to pass the name of the storage pool where the instance should be imported to. 

This PR is about providing the same functionality to Terraform, that is provided by [`incus import`](https://linuxcontainers.org/incus/docs/main/reference/manpages/incus/import/). `incus import` does allow for a flag `-storage`, which is the name of the respective storage pool. This is what we provide in the property `"pool"`

The other properties that we provide to the `device` (namely: `type = "disk" and `path = "/") are basically ignored. We just have them to make the `device` entry look the same as a regular `device` entry would look like.

An alternative would have been to define `source_file` not as a string attribute but as a `SingleNestedAttribute` and then accept a property for the name of the backup file along side with a dedicated property for the storage pool. This has been discussed in #124.

@stgraber is this explanation correct? Can you provide more context?

Reply:
Yep, that's correct.

I didn't want to make `source_file` be different for instances compared to volumes or images and since those using the instance resource are already familiar with our device syntax, it made sense to have this support a normal root device definition.